### PR TITLE
Refactor TopoDiff recipe to use diffusion framework

### PR DESCRIPTION
**Pull request by CharlelieLrt**
# PhysicsNeMo Pull Request

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/generative/topodiff/train_classifier.py [hunk 1]
```diff
@@ -64,8 +63,8 @@ def main(cfg: DictConfig) -> None:
         batch = torch.tensor(train_img[idx]).float().unsqueeze(1).to(device) * 2 - 1
```
[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply examples/generative/topodiff/train_classifier.py:61]: That's a valid point. However, the scope of this PR is only to refactor the TopoDiff recipe such that it uses the diffusion toolkit. Fixing pre-existing bugs is out-of-scope for this PR.

File: examples/generative/topodiff/utils.py [hunk 2]
```diff
@@ -127,3 +133,240 @@ def load_data_classifier(root):
...
+    ) -> Float[Tensor, " N+1"]:
+        r"""Generate discrete DDPM timesteps from ``n_steps-1`` down to ``0``."""
+        if num_steps >= self.n_steps:
+            indices = torch.arange(
+                self.n_steps - 1, -1, -1, device=device, dtype=dtype
+            )
+        else:
+            step_indices = torch.arange(num_steps, device=device, dtype=dtype)
+            scale = (self.n_steps - 1) / (num_steps - 1)
+            indices = (scale * (num_steps - 1 - step_indices)).round()
```
[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply examples/generative/topodiff/utils.py:234]: Valid point, but `num_steps == 1` is unlikely to occur in any production run

File: examples/generative/topodiff/train_classifier.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CharlelieLrt) [reply examples/generative/topodiff/train_classifier.py:61]: I'll let @mnabian chime in, but it does look like a bug to me

**Review by coreyjadams — APPROVED** [review]
approved for pyproject.toml

File: examples/generative/topodiff/train_classifier.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/generative/topodiff/train_classifier.py:61]: I agree that this is a valid point. But I suggest we leave it as is (or add a comment to the code about this issue) since the results are already validated.

**Review by mnabian — APPROVED** [review]
I think the changes look good overall. I could not spot any major issues. It would be good to track the known bugs in the code in an issue. Hopefully at some point in the future we can revisit this example and extend it to 3D cases.

**Review by pzharrington — APPROVED** [review]
